Modified 1910-1914 (AUH, ITA, TUR) to enable player choice

habbala opened the pull request:
My intent was to give the player (and AI) more choices as to how and when to start WW1. Plenty of date requirements for focuses are removed, and the assassination of Franz Ferdinand is avoidable (by accepting his proposal of trialism). The Italo-Turkish war is also modified slightly to make it more fun for both Italy and Turkey. I have also tweaked the neutral ideology to make it more easy to justify war goals and join factions. I have also experimented with the Russian revolution and making it a part of the German focus tree. AUH focus tree is a bit more predictable when it comes to inviting Italy and Romania to their faction.

habbala committed:
Modify start of ww1 (AUH focus tree)

-AUH
--If Romania accepts Bucovina, Romania joins Central Powers. Romania can be in a faction and focus is still available (stealing members is fun). If AH is not in a faction when focus is completed, it creates a new faction (Austrian Alliance).
---BUG: GER not adding ROM (if GER is faction leader for AUH)
---BUG: FRA not adding ROM (if FRA is faction leader for AUH)
--Remove focus “Call Romania to war”
--If Italy accepts Istria, Italy joins Central Powers
--Remove focus “Call Italy to war”
--Remove exclusivity between “Back Germany” and “Back Italy”
--Remove required focuses for focus “Organize the real”
--Added focus “Franz Ferdinands Proposal” to trigger the trialism proposal
---If trialism is accepted, Franz isn’t murdered in Bosnia
---If trialism is rejected, Franz is sent to Bosnia and then murdered
---Nerfed the effects of rejecting trialism proposal (-150pp -> -50pp)
--Added focus “Target Serbia” to give puppet war goal against Serbia
--Removed requirement to be at war for focus “Polnische Whermacht”
--Added focus to release Poland as a puppet
--Added focus to strike Russia directly and take polish land
--AUH now start with Bosnia annexed (since 1908)
--Nerfed AUH starting stability from 0.7 to 0.4

-GER
--Remove exclusivity between “Support Austria-Hungary” and “Support Italy”
--Added focus “Send Lenin” to boost communism in Russia with 25%
--Added focus “Arm Russian Communists” to send 10000 infantry equipment to Russian communists
---BUG: equipment isn’t removed from GER
--Added focus “Trigger Russian civil war” to trigger Russian civil war

-RUS
--Focus “Support Serbia” now also gives guarantee on Serbia

habbala committed:
Italo-Turkish war, date reqs removed

-ITA
--Focus “Italian Expansionism”
------Added spawning of 5 veteran divisions in Somaliland
--Focus “Libyan claims”
---Remove bypass if ITA and TUR is in same faction
---Remove instant state transfer from TUR to ITA
---Remove requirement to be 8/9/1911-1/1/1913
---Removed spawning of 5 veteran divisions in Tobruk
--Colonial Focus tree after “Libyan Claims”
---Added prerequisite that Italy have won Italo-Turko war

-Italo-Turkish War
--Removed white peace if either country joins a faction
--Added option for ITA to keep on fighting after taking Libya
--Removed spawning of Turkish troops in Libya

TUR
-Focus “Khedivate Egypt”
--TUR joins EGY in war against ENG
--Removed prerequisite that TUR needs to be in war against ENG

Removed date requirement for most focuses.

Neutrality
-Lowered tension limit to justify war goal from 50% to 30%
-Lowered tension limit to join faction from 50% to 40%

Democracy
-Lowered tension limit to join faction from 80% to 50%

habbala committed:
Removed date requirement for the generic part of all focus trees.

Removed date requirement for the generic part of all focus trees.

habbala committed:
Updated TODO

habbala committed:
Nerf cost of Support Equipment

nerf cost from 4 -> 2.5

habbala committed:
Modified costs for trench and forts

Buildings
-Land Fort
--Increased cost from 800 to 1400

-Coastal Fort
--Increased cost from 800 to 1000

-Trench
--Increased cost for level 1 trench from 600 to 1000
--Reduced cost for additional level from 600 to 500

habbala committed:
Zimmerman Telegram buff

If Mexico accepts Zimmerman telegram, they gain cores on Texas, New Mexico and Arizona. Otherwise it's just a shit deal, honestly.

habbala committed:
Zimmerman gives claims, not cores

Mexico gets claim on Texan, New Mexico and Arizona when accepting Zimmerman.

habbala committed:
Fixed bugged Russian CW

Fixed bug that triggered Russian CW if just one of many requirements met, not all.

habbala committed:
Fixed syntax error in "else"

habbala committed:
Modified Treaty if Central Powers win WW1

-AUH gets Venice and Zara from Italy (if italy is in entente).
-If Croatia is a puppet of AUH, then Croatia gets Montenegro (otherwise it goes to AUH)
-If Ottomans is in Central Powers, and Italy in entente, they reverse the Italo-Turkish war. If they won Italo-Turkish war, they get Eritrea and Somaliland instead.
-If Italy is in Central Powers, they get Sardinia from France.

habbala committed:
Sweden gets colonies

If Sweden is in Central Powers and wins, they may get French Caribbean via Second Treaty of Frankfurt

habbala committed:
Modified Treaty of Frankfurt (if CP win WW1)

Removed Brest-Litovsk as a prerequisite.

Italy Lost = Ethiopia gains Somaliland and Eritrea.

Italy Won = Get French and English Somaliland, Corsica, Savoie, Malta and Tunis.

If Italy won and Ottomans lost =
Get Albania.

Sweden Won =
Get French Caribbean.
If Denmark also lost, Liberate Iceland.

Mexico Won and USA Lost =
Get Texas, Arizona, New Mexico

habbala committed:
Food Mechanic

Started experimenting with a food resource, since food was one of the greatest issues and most important resource in both WW1 and WW2.

Nerfed "German Staff" idea, (20% recruitable population, instead of 50%)

habbala committed:
Added slot for food icon

habbala committed:
Fixed GUI for food, updated TODO

Also removed AUH getting a state they most probably already owned from Italy, if they win WW1.

habbala committed:
Added farm building

Farm building may be built to get food.

habbala committed:
Fixed errors, bugs and typos

habbala committed:
Fixed bugs where Bosnia and Asir remained in WW1, without existing

habbala committed:
Moved trialism in AUH focus tree

habbala committed:
Made treaty of Brest-Litovsk optional

Created an event for germaany to accept or refuse the treaty of brest litovsk. Later improvement would require more war participants to choose as well. Fixed bugs in Russian Civil War. Sweden now gets Finland if they are in CP via Brest-Litovsk.

habbala committed:
Remade Russian civil war

Remade russian civil war into an actual civil war, making it much more flexible and less laggy. Also seperated tsars abdication into its own event, making it possible that he doesn't abdicates, but civil war still breaks out later.

NicolasGrosjean commented on mod/thegreatwar/Todo.md:
```diff
@@ -41,3 +41,4 @@ BUGS
 -Event that says "Russia declares war on Austria" triggers 8/9/1915, even if they have already lost WW1. War isn't started however
 -France declares war on Germany 19/9/1915, even if France just lost WW1
 -Choosing to sign the treaty of Brest-Litovsk as Germany does not actually trigger the event
+-Main menu background is from DLC?
```
@habbala 
Yes I believe because I have no DLC and I have gfx\loadingscreens\load_tfv.dds from mod as main menu backround

habbala replied:
Hi Nicolas! So glad someone is still on here! Have you tried my branch? I'm curious what people think of my changes :)

NicolasGrosjean replied:
I have not tested yet.

The most important is that @Wolferos tests it because I think he is the only one who can accept pull requests. With the 11th November, he will surely come.

habbala committed:
Manpower laws affect production

Manpower laws now affect production output in a larger way. Tried to balance this by removing the penalty to training time.

Factory output goes:
+15%
+0%
-5%
-15%
-30%
-50%

MasterWaifuman commented on mod/thegreatwar/Todo.md:
```diff
@@ -0,0 +1,54 @@
+
+Habbalas TODO
+
+-TREATIES
```
I am currently working on making treaties decisions. This will be more accurate and less buggy. Moreover, it makes possible to ignore them if player wants default peace conference. Now I already completed start of WW and going to make decision of interventions for countries that joined war later.

MasterWaifuman commented on mod/thegreatwar/Todo.md:
```diff
@@ -0,0 +1,54 @@
+-Add mechanic where taking/holding non core states lowers stability
+-Add mechanic where war support decreases for every war participant every year (example event "Spanish flu")
+-Lower PP cost for influencing foreign ideologies (ATM it costs 5PP!)
+-Add mechanic for country specific guarantees. Such as "Russia guarantees Serbia against Austrian agression". Gives Russia the choice to intervene or not if Austria attacks Serbia. Similar to Rhineland, Memel, Danzig events in base game.
```
And again I added decision that allows player to intervene or not (but get some penalties such as war support reduction, manpower and pp reduction but increase in stability.

MasterWaifuman commented on mod/thegreatwar/Todo.md:
```diff
@@ -0,0 +1,54 @@
+BUGS
+-Remove errors
+-German focus "Invite Sweden" will ask the player (Germany) to accept or deny (not Sweden)
+-If a democratic nation (Sweden, Mexico, Romania, Italy) joins Central Powers via focus tree, they will leave immediately
```
This is not really necessary. Although, it may be possible to make civil war of democrats against authoritarians as a result of democratic country joining Central Powers.

MasterWaifuman commented on mod/thegreatwar/Todo.md:
```diff
@@ -0,0 +1,54 @@
+-If a democratic nation (Sweden, Mexico, Romania, Italy) joins Central Powers via focus tree, they will leave immediately
+-Second treaty of Frankfurt does not cede any Italian states/colonies, even if Italy was in entente.
+-Event that says "Russia declares war on Austria" triggers 8/9/1915, even if they have already lost WW1. War isn't started however
+-France declares war on Germany 19/9/1915, even if France just lost WW1
```
2 above problems solved with decisions.
